js/packages/common/src/contexts/connection.tsx

```diff
@@ -780,71 +783,68 @@ async function awaitTransactionSignatureConfirmation(
     err: null,
   };
   let subId = 0;
-  status = await new Promise(async (resolve, reject) => {
+  status = await (async (): Promise<SignatureStatus | null | void> => {
```

Comment:
This is the only part of the old https://github.com/metaplex-foundation/metaplex/pull/1076 that I wanted some extra eyes on. Most of the other changes are non-controversial, but this one has the potential to actually change the runtime behavior.

Can you give this a once-over and make sure that it still does what it's supposed to.

Reply:
I'll give it a look later today! Thanks :)

Reply:
Reviewed everything. I think this looks fine, but I'm going to give it the full test first and hopefully merge this in soon.